v2/internal/reconcilers/arm/azure_generic_arm_reconciler_instance.go

```diff
@@ -284,20 +312,54 @@ func (r *azureDeploymentReconcilerInstance) BeginCreateOrUpdateResource(ctx cont
 	r.Recorder.Eventf(r.Obj, v1.EventTypeNormal, string(CreateOrUpdateActionBeginCreation), "Successfully sent resource to Azure with ID %q", armResource.GetID())
 
 	// If we are done here it means the deployment succeeded immediately. It can't have failed because if it did
-	// we would have taken the err path above.
+	// we would have taken the error path above.
 	if pollerResp.Poller.Done() {
 		return r.handleCreatePollerSuccess(ctx)
 	}
 
 	resumeToken, err := pollerResp.Poller.ResumeToken()
 	if err != nil {
-		return ctrl.Result{}, errors.Wrapf(err, "couldn't create PUT resume token for resource %q", armResource.GetID())
+		return ctrl.Result{},
+			errors.Wrapf(err, "couldn't create PUT resume token for resource %q", armResource.GetID())
 	}
 
 	SetPollerResumeToken(r.Obj, pollerResp.ID, resumeToken)
 	return ctrl.Result{Requeue: true}, nil
 }
 
+func (r *azureDeploymentReconcilerInstance) preReconciliationCheck(ctx context.Context) (extensions.PreReconcileCheckResult, error) {
+	// Create a checker for access to the extension point, if required
+	checker, extensionFound := extensions.CreatePreReconciliationChecker(r.Extension)
+	if !extensionFound {
+		// No extension found, nothing to do
+		return extensions.ProceedWithReconcile(), nil
+	}
+
+	// Having a checker requires our resource to have an up-to-date status
+	r.Log.V(Verbose).Info("Refreshing Status of resource")
+	statusErr := r.updateStatus(ctx)
+	if statusErr != nil && !genericarmclient.IsNotFoundError(statusErr) {
+		// We have an error, and it's not because the resource doesn't exist yet
+		return extensions.PreReconcileCheckResult{}, statusErr
+	}
+
+	// We also need to have our owner, it too with an up-to-date status
+	owner, ownerErr := r.ResourceResolver.ResolveOwner(ctx, r.Obj)
```

Comment:
minor: Wondering if it's better to leave this to the `checker` to choose to do if it wants? Rather than passing owner you could pass `ResourceResolver` - then the checker could decide to call it, or not if it didn't need it.

That would reduce unneeded owner lookups -- although I do think they're cached so we're probably ok either way.